### DISCUSSION TITLE
Conditional reinit(x, value), if x is state variable

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#13339] [Q&A] [answered]
Hi, 
Is there a way to detect if a derivative variable, x, has been selected as a state to apply that condition to the execution of reinit(x, value)? I am developing a "parametric" simulation (including also stochastic variables) inside the own model through classes that automate the reinit of the model, and this capability could be very useful. 

Thanks in advance!

**Comment by mpradovelasco** [comment]
An alternative is that reinit(x, value) only acts on state variables, leaving other variable types intact. If this solution could be implemented while respecting the Modelica standard, it would be glorious.

**Reply by casella** [reply, marked as answer]
See #13247. According to the Modelica Language Specification, reinit(x, <something>) forces x to be a state variable. 

The problem is now fixed in the nightly build.

**Reply by mpradovelasco** [reply]
Thank you @casella
